renga/cli/notebooks.py

```diff
@@ -94,11 +98,44 @@ def launch(ctx, config, engine, endpoint):
             command="start-notebook.sh --NotebookApp.token={0}".format(
                 notebook_token),
             ports=['8888'],
-            image='jupyter/minimal-notebook:latest',
+            image='rengahub/minimal-notebook:latest',
             labels=['renga.notebook.token={0}'.format(notebook_token)],
             endpoint=endpoint)
         cfg = ctx.obj['config'].get('project', config)['endpoints'][endpoint]
         cfg['notebook'] = context.id
 
-    execution = context.run(engine=engine)
-    click.echo(execution.url)
+    execution = context.run(
+        engine=engine,
+        environment={
```

Comment:
the code used in an execution should be a part of the context not the execution; these should be a part of, and come from, the context spec

Reply:
It will be refactored one we have an code node connecting to the context.